public/app/features/alerting/unified/components/notification-policies/Policy.tsx

```diff
@@ -0,0 +1,660 @@
+  onShowAlertInstances: (alertGroups: AlertmanagerGroup[], matchers?: ObjectMatcher[]) => void;
+}
+
+const Policy: FC<PolicyComponentProps> = ({
+  receivers = [],
+  contactPointsState,
+  readOnly = false,
+  alertGroups = [],
+  alertManagerSourceName,
+  currentRoute,
+  routeTree,
+  inheritedProperties,
+  routesMatchingFilters = [],
+  onEditPolicy,
+  onAddPolicy,
+  onDeletePolicy,
+  onShowAlertInstances,
+}) => {
+  const styles = useStyles2(getStyles);
+  const isDefaultPolicy = currentRoute === routeTree;
+
+  const permissions = getNotificationsPermissions(alertManagerSourceName);
+  const canEditRoutes = contextSrv.hasPermission(permissions.update);
+  const canDeleteRoutes = contextSrv.hasPermission(permissions.delete);
+
+  const contactPoint = currentRoute.receiver;
+  const continueMatching = currentRoute.continue ?? false;
+  const groupBy = currentRoute.group_by ?? [];
+  const muteTimings = currentRoute.mute_time_intervals ?? [];
+  const timingOptions: TimingOptions = {
+    group_wait: currentRoute.group_wait,
+    group_interval: currentRoute.group_interval,
+    repeat_interval: currentRoute.repeat_interval,
+  };
+
+  const matchers = normalizeMatchers(currentRoute);
+  const hasMatchers = Boolean(matchers && matchers.length);
+  const hasMuteTimings = Boolean(muteTimings.length);
+  const hasFocus = routesMatchingFilters.some((route) => route.id === currentRoute.id);
+
+  // gather warnings and errors here
+  const warnings: ReactNode[] = [];
+  const errors: ReactNode[] = [];
+
+  // if the route has no matchers, is not the default policy (that one has none) and it does not continue
+  // then we should warn the user that it's a suspicious setup
+  if (!hasMatchers && !isDefaultPolicy && !continueMatching) {
+    warnings.push(wildcardRouteWarning);
+  }
+
+  // if the receiver / contact point has any errors show it on the policy
+  const actualContactPoint = contactPoint ?? inheritedProperties?.receiver ?? '';
+  const contactPointErrors = contactPointsState ? getContactPointErrors(actualContactPoint, contactPointsState) : [];
+
+  contactPointErrors.forEach((error) => {
+    errors.push(error);
+  });
+
+  const childPolicies = currentRoute.routes ?? [];
+  const hasChildPolicies = Boolean(childPolicies.length);
+  const isGrouping = Array.isArray(groupBy) && groupBy.length > 0;
+  const hasInheritedProperties = inheritedProperties && Object.keys(inheritedProperties).length > 0;
+
+  const isEditable = canEditRoutes;
+  const isDeletable = canDeleteRoutes && !isDefaultPolicy;
+
+  const matchingAlertGroups = useMemo(
+    () => findMatchingAlertGroups(routeTree, currentRoute, alertGroups),
+    [alertGroups, currentRoute, routeTree]
+  );
+
+  // sum all alert instances for all groups we're handling
+  const numberOfAlertInstances = sumBy(matchingAlertGroups, (group) => group.alerts.length);
+
+  // TODO dead branch detection, warnings for all sort of configs that won't work or will never be activated
+  return (
```

Comment:
Should we add more tailor-made components into this one?
The number of conditions and branches make the logic quite difficult to follow so maybe limiting the amount of UI code would help
e.g. Instead of 
```
{errors.length > 0 && (
                <HoverCard
                  arrow
                  placement="top"
                  content={
                    <Stack direction="column" gap={0.5}>
                      {errors.map((error) => (
                        <Fragment key={uniqueId()}>{error}</Fragment>
                      ))}
                    </Stack>
                  }
                >
                  <span>
                    <Badge icon="exclamation-circle" color="red" text={pluralize('error', errors.length, true)} />
                  </span>
                </HoverCard>
              )}
```
have sth like
```
{errors.length > 0 && <PolicyErrors errors={errors> />}

Reply:
Yea I totally agree, I haven't spent enough time figure out what abstractions we can make here – will try to clean this up a bit.

Reply:
I've cleaned up the policy component – right now the sub components still live in the same file (since it's very likely that if we change the policy component those will have to be updated too, I prefer to keep code that changes often together in the same file.